assets/src/components/d3/createResourceAccessChart.js

```diff
@@ -66,36 +65,78 @@ function appendLegend (svg) {
 
   legendRect
     .attr('y', (_, i) => legendY + i * legendInterval)
-    .attr('x', w)
+    .attr('x', 0)
     .style('fill', d => d[1])
 }
 
+function truncate (selection, labelWidth) {
+  selection.each(function (d) {
+    const textEl = d3.select(this)
+    const textString = d.split('|')[1]
+    textEl.text(textString)
+    const textPixelLength = textEl.node().getComputedTextLength()
+    if (textPixelLength > labelWidth) {
+      const numChars = Math.floor(labelWidth / textPixelLength * textString.length) - 3
+      textEl.text(`${textString.substring(0, numChars - 1)}...`)
+    } else {
+      textEl.text(textString)
+    }
+  })
+}
+
 function createResourceAccessChart ({ data, width, height, domElement }) {
   const resourceData = data.sort((a, b) => b.total_percent - a.total_percent)
 
-  const [mainWidth, miniHeight] = adjustViewport(width, height, mainMargin)
+  const sideMarginSize = width * 0.075
+  const margin = { top: 0, right: sideMarginSize, bottom: 0, left: sideMarginSize }
+
+  const [availWidth, availHeight] = adjustViewport(width, height, margin)
+
+  /*
+  The decimal multipliers for mainMargin.left, mainWidth, mainMargin.right, miniMargin.left,
+  miniWidth, and miniMargin.right should add up to 1.0. In other words, the sum of those values
+  should be equal to the availWidth.
+  */
+  const mainWidth = availWidth * 0.55
+  const mainHeight = availHeight * 0.7
+  const mainMargin = {
+    top: availHeight * 0.15,
+    right: 0,
+    bottom: availHeight * 0.15,
+    left: availWidth * 0.225
+  }
+  const resourceLabelWidth = mainMargin.left * 0.889 - foreignObjSide
```

Comment:
I understand the comment for the values above but not sure about this calclulation. 
`mainMargin.left * 0.889 - 24 ` (currently defined somewhere else). What's the 0.889?

Reply:
Yeah, this isn't ideal. Essentially, I want to find 8/9's of `mainMargin.left`, since that would be `0.20`, leaving the same margin size as `miniMargin.left`. It's a bit awkward because the mainMargin.left is really acting like it's own section. I can fix this so that I define the variable after I define the `mini` stuff, and then use `const resourceLabelWidth = mainMargin.left - miniMargin.left - foreignObjSide.` Hopefully, that will be clearer.